log: log everything to a single file

knz opened the pull request:
This patch removes the split between INFO/ERROR/WARNING logs; all the
logs are emitted to the same file.
The auto-rotate behavior is preserved.

Fixes #13905.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/14347)
<!-- Reviewable:end -->

tamird commented:
Reviewed 1 of 13 files at r1.
Review status: 1 of 15 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[pkg/server/serverpb/status.proto, line 146 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14347#-KfwoPj6kVEb1TwEXo6H:-KfwoPj6kVEb1TwEXo6I:b-kkhtb) ([raw file](https://github.com/cockroachdb/cockroach/blob/8303a40a8406413415dbe9fd1ceb7e1c056c848c/pkg/server/serverpb/status.proto#L146)):*
> ```protobuf
>   string max = 5;
>   string pattern = 6;
>   // level is not used any more, but we must keep it because it was not
> ```

this is incorrect; all fields are optional in proto 3.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14347)*
<!-- Sent from Reviewable.io -->

bdarnell commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

---

Reviewed 12 of 13 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[pkg/ui/yarn.lock, line 1 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/14347#-KfwpPHCCm9YNKyriCeB:-KfwpPHCCm9YNKyriCeC:bjn8sky) ([raw file](https://github.com/cockroachdb/cockroach/blob/954c314def1a9ce61f54b43af4a006dd89ba32fb/pkg/ui/yarn.lock#L1)):*
> ```
> # THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
> ```

The changes here look unintentional.

---

*[pkg/util/log/clog.go, line 974 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/14347#-KfwpvTWzGQOwWQVHyew:-KfwpvTWzGQOwWQVHyex:b-b3iw1o) ([raw file](https://github.com/cockroachdb/cockroach/blob/954c314def1a9ce61f54b43af4a006dd89ba32fb/pkg/util/log/clog.go#L974)):*
> ```Go
> const bufferSize = 256 * 1024
> 
> func (l *loggingT) closeFilesLocked() error {
> ```

This function can be renamed to singular `File`

---

*[pkg/util/log/clog.go, line 993 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/14347#-Kfwq3DMPDPD_-bvQXUQ:-Kfwq3DMPDPD_-bvQXUR:b-a7q0bx) ([raw file](https://github.com/cockroachdb/cockroach/blob/954c314def1a9ce61f54b43af4a006dd89ba32fb/pkg/util/log/clog.go#L993)):*
> ```Go
> 		sb := &syncBuffer{
> 			logger: l,
> 			sev:    Severity_INFO,
> ```

Can the `syncBuffer.sev` field be removed now?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14347#-:-Kfwr3Al4Arzyst8YAOt:bg7igkv)*
<!-- Sent from Reviewable.io -->

knz committed:
log: log everything to a single file

This patch removes the split between INFO/ERROR/WARNING logs; all the
logs are emitted to the same file.
The auto-rotate behavior is preserved.

knz commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[pkg/server/serverpb/status.proto, line 146 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14347#-KfwoPj6kVEb1TwEXo6H:-Kfx6qFI4QtdUmTmjBPV:b-9ym445) ([raw file](https://github.com/cockroachdb/cockroach/blob/8303a40a8406413415dbe9fd1ceb7e1c056c848c/pkg/server/serverpb/status.proto#L146)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

this is incorrect; all fields are optional in proto 3.
</blockquote></details>

Thanks! This way I do not need to modify the protos nor regenerate embedded.go.

---

*[pkg/ui/yarn.lock, line 1 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/14347#-KfwpPHCCm9YNKyriCeB:-Kfx6uJ536okRpOzkIJk:bo6apac) ([raw file](https://github.com/cockroachdb/cockroach/blob/954c314def1a9ce61f54b43af4a006dd89ba32fb/pkg/ui/yarn.lock#L1)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

The changes here look unintentional.
</blockquote></details>

Indeed.

---

*[pkg/util/log/clog.go, line 974 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/14347#-KfwpvTWzGQOwWQVHyew:-Kfx74MVVs1cJ-CWoVFl:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/954c314def1a9ce61f54b43af4a006dd89ba32fb/pkg/util/log/clog.go#L974)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

This function can be renamed to singular `File`
</blockquote></details>

Done.

---

*[pkg/util/log/clog.go, line 993 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/14347#-Kfwq3DMPDPD_-bvQXUQ:-Kfx7VGdb74RmP42FBQm:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/954c314def1a9ce61f54b43af4a006dd89ba32fb/pkg/util/log/clog.go#L993)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

Can the `syncBuffer.sev` field be removed now?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14347)*
<!-- Sent from Reviewable.io -->

knz commented:
TFYRs

tamird commented:
Reviewed 11 of 11 files at r3.
Review status: all files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

*[pkg/util/log/clog.go, line 944 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/14347#-KfxCkqrkrJQvyaQMFfL:-KfxCkqsTJ7rkiugtJ8B:b-hd77ar) ([raw file](https://github.com/cockroachdb/cockroach/blob/4855fc87eb04e7055b5781897aedc21cff77de8d/pkg/util/log/clog.go#L944)):*
> ```Go
> 	} {
> 		buf := formatLogEntry(Entry{
> 			Severity:  Severity_INFO,
> ```

why are we still including the severity here? I would just mark the `severity` field in the proto "reserved" and remove this.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14347)*
<!-- Sent from Reviewable.io -->
